src/foundry/common/utils/helpers.mjs.d.ts

```diff
@@ -253,12 +274,22 @@ export declare function invertObject<T extends Record<string | number | symbol,
 export declare function isNewerVersion(v1: number | string, v0: number | string): boolean;
 
 /**
- * A simple function to test whether or not an Object is empty
+ * A simple function to test whether an Object is empty
  * @param obj - The object to test
  * @returns Is the object empty?
+ * @deprecated since v10, will be removed in v11 - Use isEmpty instead.
  */
 export declare function isObjectEmpty(obj: object): boolean;
 
+/**
+ * Test whether a value is empty-like; either undefined or a content-less object.
+ * @param value - The value to test
+ * @returns Is the value empty-like?
+ */
+export function isEmpty(
+  value: undefined | unknown[] | Record<string, unknown> | Set<unknown> | Map<unknown, unknown>
```

Comment:
I think this also can deal with objects that have number keys, and they don't need to have an index signature.